goshopify.go

```diff
@@ -205,6 +210,12 @@ func NewClient(app App, shopName, token string) *Client {
 	return c
 }
 
+// WithLogger just sets logger and returns the Client.
+func (c *Client) WithLogger(l Logger) *Client {
```

Comment:
Another way to do this is to do WithContext(). Add the api limit info into the context which can be logged along with client information each time Do() is called